crates/frontend/src/components/button.rs

```diff
@@ -7,18 +7,35 @@ pub fn button<F: Fn(MouseEvent) + 'static>(
     on_click: F,
     #[prop(default = String::new())] class: String,
     #[prop(default = String::new())] id: String,
+    #[prop(default = false)] loading: bool,
 ) -> impl IntoView {
     view! {
         <button
-            class=format!(
+            class=if loading {
+                format!(
+                    "btn-purple hover:cursor-not-allowed font-medium rounded-lg text-sm px-5 py-2.5 text-center me-2 mb-2 {class}",
+                )
+            }
+            else {
+                format!(
                 "btn-purple font-medium rounded-lg text-sm px-5 py-2.5 text-center me-2 mb-2 {class}",
             )
```

Comment:
Can just simplify this tailwind, only writing the uncommon ones inside condition